### PR TITLE
init feature/AP5-119

### PR DESCRIPTION
**Pull request by jasonephraim**

### CODE REVIEW

**Review by mmevans — APPROVED** [review]

File: server/src/graphql/resolvers/deliveryPhase.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+const groupByType = (parent) => {
+  return parent.reduce((result, currentValue) => {
+    (result[currentValue.typeId] = result[currentValue.typeId] || []).push(
+      currentValue
+    );
+    return result;
+  }, {});
+};
+
+// schema will not accept spaces or special characters as type field names so we must replace versus a lookup (which would return strings with spaces and special characters)
```
Comment (jeremy-roussel) [inline server/src/graphql/resolvers/deliveryPhase.js:12]: Can you not just make a copy of the object without the hidden attributes, and use the copy to update?

File: server/src/graphql/resolvers/deliveryPhase.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+const keyReplace = (contentObject) => {
+  const preDeliveryOldKey = '52d02965-ca2e-4b2a-8bae-47bedb54a473';
+  const deliveryOldKey = 'cd52d508-0bca-4c9d-afed-f9039e1ac0f0';
+  if (preDeliveryOldKey !== 'preDeliveryPhase') {
```
Comment (jeremy-roussel) [inline server/src/graphql/resolvers/deliveryPhase.js]: This will always be false?

File: server/src/graphql/resolvers/deliveryPhase.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    Object.defineProperty(
+      contentObject,
+      'preDeliveryPhase',
+      Object.getOwnPropertyDescriptor(contentObject, preDeliveryOldKey)
+    );
+    delete contentObject[preDeliveryOldKey];
+  }
+  if (deliveryOldKey !== 'deliveryPhase') {
```
Comment (jeremy-roussel) [inline server/src/graphql/resolvers/deliveryPhase.js]: This too?

File: server/src/graphql/resolvers/deliveryPhase.js (hunk 2 above)
Comment (jasonephraim) [reply server/src/graphql/resolvers/deliveryPhase.js]: fixed - removed unnecessary conditional

File: server/src/graphql/resolvers/deliveryPhase.js (hunk 3 above)
Comment (jasonephraim) [reply server/src/graphql/resolvers/deliveryPhase.js]: fixed - removed unnecessary conditional

**Review by jeremy-roussel — APPROVED** [review]

File: server/src/utils/adjustFollowingDeliveryPhases.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          newModel.itemInFrontOf = nextPhase.itemInFrontOf;
+          nextUnfrozenPhase(newModel);
+        } else {
+          nextPhase.itemInFrontOf = newModel.id;
+          await db.deliveryPhase.upsert(nextPhase.dataValues);
+          newModel.itemInFrontOf = null;
+          await db.deliveryPhase.upsert(newModel);
+        }
+      }
+      if (!nextPhase.freezeDuration) {
```
Comment (BarretteBanner) [inline server/src/utils/adjustFollowingDeliveryPhases.js]: Wouldn't this be an else to your first if

File: server/src/utils/adjustFollowingDeliveryPhases.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        if (newModel.durationInHours >= nextPhase.durationInHours) {
+          await db.deliveryPhase.destroy({
+            where: nextPhase.dataValues,
+          });
+        } else {
+          const newNextPhaseDuration =
+            nextPhase.durationInHours - newModel.durationInHours;
```
Comment (BarretteBanner) [inline server/src/utils/adjustFollowingDeliveryPhases.js:27]: For stuff like this I think it's actually more readable to do it in one line and get rid of the const

**Review by BarretteBanner — CHANGES_REQUESTED** [review]
Please address comments

File: server/src/utils/adjustFollowingDeliveryPhases.js (hunk 4 above)
Comment (jasonephraim) [reply server/src/utils/adjustFollowingDeliveryPhases.js]: fixed

File: server/src/utils/adjustFollowingDeliveryPhases.js (hunk 5 above)
Comment (jasonephraim) [reply server/src/utils/adjustFollowingDeliveryPhases.js:27]: removed const

**Review by BarretteBanner — APPROVED** [review]

**Pull request merged**